File: risc0/zkp/core/bench/benchmark.cpp
// Copyright 2022 Risc0, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include "benchmark/benchmark.h"

#include "risc0/core/rng.h"
#include "risc0/zkp/core/ntt.h"

using namespace risc0;

static void BM_Interpolate_NTT(benchmark::State& state) {
  size_t n = state.range(0);
  size_t size = 1 << n;

  std::vector<Fp> buf(size);
  PseudoRng rng(2);
  for (size_t i = 0; i < size; i++) {
    buf[i] = Fp::random(rng);
  }

  for (auto _ : state) {
    interpolateNTT(buf.data(), size);
  }
}

BENCHMARK(BM_Interpolate_NTT)->Unit(benchmark::kMillisecond)->Arg(10)->Arg(15)->Arg(20);

BENCHMARK_MAIN();

File: examples/ecdsa/k256/src/bin/benchmark.rs
// Copyright 2025 RISC Zero, Inc.
//
// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
// http://opensource.org/licenses/MIT>, at your option. This file may not be
// copied, modified, or distributed except according to those terms.
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// SPDX-License-Identifier: Apache-2.0 OR MIT

use k256_methods::BENCHMARK_ELF;
use risc0_zkvm::{ExecutorEnv, default_executor};

// Simple main to load and run the benchmark binary in the RISC Zero Executor.
fn main() {
    let env = ExecutorEnv::builder().build().unwrap();
    let exec = default_executor();
    std::hint::black_box(exec.execute(env, BENCHMARK_ELF)).unwrap();
}
